src/BenchmarkDotNet/Toolchains/MonoWasm/WasmAppBuilder.cs

```diff
@@ -0,0 +1,201 @@
+        Directory.CreateDirectory(outputDir);
+        Directory.CreateDirectory(Path.Combine(outputDir, "managed"));
+        foreach (var assembly in _assemblies.Values)
+            File.Copy(assembly.Location, Path.Combine(outputDir, "managed", Path.GetFileName(assembly.Location)), true);
+        foreach (var f in new string[] { "dotnet.wasm", "dotnet.js" })
+            File.Copy(Path.Combine(appDir, f), Path.Combine(outputDir, f), true);
+        File.Copy(WasmSettings.WasmMainJS, Path.Combine(outputDir, "runtime.js"),  true);
+
+        string supportFilesDir = Path.Combine(outputDir, "supportFiles");
+        Directory.CreateDirectory(supportFilesDir);
+
+        var filesToMap = copySystemPrivateCoreLib(appDir, supportFilesDir);
+
+        using (var sw = File.CreateText(Path.Combine(outputDir, "mono-config.js")))
+        {
+            sw.WriteLine("config = {");
+            sw.WriteLine("\tvfs_prefix: \"managed\",");
+            sw.WriteLine("\tdeploy_prefix: \"managed\",");
+            sw.WriteLine("\tenable_debugging: 0,");
+            sw.WriteLine("\tassembly_list: [");
+            foreach (var assembly in _assemblies.Values)
+            {
+                sw.Write("\t\t\"" + Path.GetFileName(assembly.Location) + "\"");
+                sw.WriteLine(",");
+            }
+            sw.WriteLine ("\t],");
+            sw.WriteLine("\tfiles_to_map: [");
+            foreach (KeyValuePair<string, List<string>> keyValuePair in filesToMap)
+            {
+                sw.WriteLine("\t{");
+                sw.WriteLine($"\t\tdirectory: \"{keyValuePair.Key}\",");
+                sw.WriteLine("\t\tfiles: [");
+                foreach (string file in keyValuePair.Value)
+                {
+                    sw.WriteLine($"\t\t\t\"{file}\",");
+                }
+                sw.WriteLine("\t\t],");
+                sw.WriteLine("\t},");
+            }
+            sw.WriteLine ("\t],");
+            sw.WriteLine ("}");
+        }
+
+
+        using (var sw = File.CreateText(Path.Combine(outputDir, "run-v8.sh")))
+        {
+            sw.WriteLine("v8 --expose_wasm runtime.js -- --run " + Path.GetFileName(mainAssemblyPath) + " $*");
+        }
+
+        return true;
+    }
+
+    private void Add(MetadataLoadContext mlc, Assembly assembly)
+    {
+        if (_assemblies.ContainsKey(assembly.GetName().Name))
+            return;
+        _assemblies[assembly.GetName().Name] = assembly;
+
+
+        foreach (var aname in assembly.GetReferencedAssemblies())
+        {
+            try
+            {
+                var refAssembly = mlc.LoadFromAssemblyName(aname);
+                Add(mlc, refAssembly);
+            }
+            catch (System.IO.FileNotFoundException)
+            {
+                Console.Error.WriteLine($"WARNING: Could not load {aname.Name}");
+            }
+        }
+    }
+
+    private Dictionary<string, List<string>> copySystemPrivateCoreLib(string appDir, string supportFilesDir)
```

Comment:
I don't think this is needed or does BDN try to open System.Private.CoreLib.dll from the filesystem?

Reply:
It does, which is why we added it to the file system in the first place. I never did go and understand why it tries to do that though; maybe fixable on the BDN end?

Reply:
It turns out not to be needed any more due to other changes, so I got rid of it.